starknet-crypto/src/rfc6979.rs

```diff
@@ -0,0 +1,121 @@
+use crate::FieldElement;
+
+use crypto_bigint::{ArrayEncoding, ByteArray, Integer, U256};
+use ff::PrimeField;
+use hmac::digest::{BlockInput, FixedOutput, Reset, Update};
+use zeroize::{Zeroize, Zeroizing};
+
+const EC_ORDER: U256 =
+    U256::from_be_hex("0800000000000010ffffffffffffffffb781126dcae7b2321e66a241adc64d2f");
+
+/// Deterministically generate ephemeral scalar `k` based on RFC 6979.
+///
+/// ### Arguments
+///
+/// * `message_hash`: message hash
+/// * `private_key`: private key
+/// * `seed`: extra seed for additional entropy
+pub fn generate_k(
+    message_hash: &FieldElement,
+    private_key: &FieldElement,
+    seed: Option<&FieldElement>,
+) -> FieldElement {
+    // The message hash padding as implemented in `cairo-lang` is not needed here. The hash is
+    // padded in `cairo-lang` only to make sure the lowest 4 bits won't get truncated, but here it's
+    // never getting truncated anyways.
+    let message_hash = U256::from_be_slice(&message_hash.to_repr().0).to_be_byte_array();
+    let private_key = crypto_bigint::U256::from_be_slice(&private_key.to_repr().0);
```

Comment:
nit: `U256` doesn't need to be fully qualified

Reply:
Yes you're right. Missed it when cleaning up junk code lol.